apps/web/pages/_app.js

```diff
@@ -8,15 +8,28 @@ import {
   UserContextProvider,
   SearchContextProvider,
 } from '@kiwicom/margarita-core';
+import qs from 'qs';
 
 export default class App extends NextApp {
+  getContextFromURL(params: Object): Object {
+    return Object.keys(params).length
+      ? {
+          ...params,
+          travelFrom: Object.values(qs.parse(params.travelFrom)),
+          travelTo: Object.values(qs.parse(params.travelTo)),
+        }
+      : {};
+  }
+
   render() {
-    const { Component, pageProps } = this.props;
+    const { Component, pageProps, router } = this.props;
     return (
       <Container>
         <UserContextProvider>
           <AlertContextProvider>
-            <SearchContextProvider>
+            <SearchContextProvider
```

Comment:
I don't think this should be done on each page. It should only be done on the results page.

Reply:
We did it this way as it is how it's done on the `mobile` version. Also we need it on `Search` and `Results`. I have now added a check to only do this if there is `router.query`

Reply:
Sorry, I was specific enough here. 

I am ok with having the provider here. But maybe the provider should have a method, setStateFromQueryParams or something like that, and we just call that method from the search and result screen.

Reply:
That looks great, putting it as an action, will do it now